server/channels/api4/report.go

```diff
@@ -153,3 +155,95 @@ func fillUserReportOptions(values url.Values) (*model.UserReportOptions, *model.
 		SearchTerm:   values.Get("search_term"),
 	}, nil
 }
+
+// getPostsForReporting retrieves posts for reporting purposes with cursor-based pagination.
+//
+// API Endpoint: POST /api/v4/reports/posts
+//
+// Cursor Behavior:
+//   - The cursor is opaque and self-contained (base64-encoded)
+//   - When a cursor is provided, it contains all query parameters from the initial request
+//   - Query parameters in the request body (time_field, sort_direction, include_deleted, exclude_system_posts)
+//     are IGNORED when a cursor is present - the cursor's parameters take precedence
+//   - This allows clients to keep sending the same parameters on every request without causing errors
+//   - For the first page or to start a new query, omit the cursor or send an empty string
+//
+// Required permissions: System Admin (PERMISSION_MANAGE_SYSTEM)
+func getPostsForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
+	// Require system admin permission for accessing posts reporting
+	if !c.IsSystemAdmin() {
+		c.SetPermissionError(model.PermissionManageSystem)
+		return
+	}
+
+	// Request body contains both options and cursor
+	var request struct {
+		model.ReportPostOptions
+		model.ReportPostOptionsCursor
+	}
+	if jsonErr := json.NewDecoder(r.Body).Decode(&request); jsonErr != nil {
+		c.SetInvalidParamWithErr("body", jsonErr)
+		return
+	}
+
+	// Validate required parameters
+	if request.ChannelId == "" {
+		c.SetInvalidParam("channel_id")
+		return
+	}
+	if !model.IsValidId(request.ChannelId) {
+		c.SetInvalidParam("channel_id")
+		return
+	}
+
+	// Verify channel exists before querying posts
+	// This provides a better error message than returning an empty result set
+	channel, appErr := c.App.GetChannel(c.AppContext, request.ChannelId)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+	if channel == nil {
+		c.Err = model.NewAppError("getPostsForReporting", "api.post.get_posts_for_reporting.channel_not_found", nil, fmt.Sprintf("channel_id=%s", request.ChannelId), http.StatusNotFound)
+		return
+	}
```

Comment:
Can we handle the empty result set instead of having this additional query? This query always runs, but asking for an invalid channel ID seems like a not so frequent case.